larpandoracontent/LArControlFlow/TrackDirectionTool.h

```diff
@@ -0,0 +1,1089 @@
+{
+    return m_forwardschisquared;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::DirectionFitObject::GetBackwardsChiSquared()
+{
+    return m_backwardschisquared;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void TrackDirectionTool::DirectionFitObject::SetForwardsChiSquared(float forwardsChiSquared)
+{
+    m_forwardschisquared = forwardsChiSquared;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void TrackDirectionTool::DirectionFitObject::SetBackwardsChiSquared(float backwardsChiSquared)
+{
+    m_backwardschisquared = backwardsChiSquared;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::DirectionFitObject::GetForwardsChiSquaredPerHit()
+{
+    return m_forwardschisquared/m_nhits;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::DirectionFitObject::GetBackwardsChiSquaredPerHit()
+{
+    return m_backwardschisquared/m_nhits;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void TrackDirectionTool::DirectionFitObject::SetNHits(float numberHits)
+{
+    m_nhits = numberHits;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline int TrackDirectionTool::DirectionFitObject::GetNHits()
+{
+    return m_hitchargevector.size();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline int TrackDirectionTool::DirectionFitObject::GetDirectionEstimate()
+{
+    return (m_forwardschisquared <= m_backwardschisquared ? 1 : 0);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::DirectionFitObject::GetMinChiSquared()
+{
+    return std::min(m_forwardschisquared, m_backwardschisquared);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::DirectionFitObject::GetMinChiSquaredPerHit()
+{
+    return (m_nhits != 0 ? std::min(m_forwardschisquared, m_backwardschisquared)/m_nhits : std::numeric_limits<float>::max());
```

Comment:
This was set by Joris - I think it should be fine, but I could change it to 45 and it would have the same effect I think (as this would push it outside the probability boundary to a default value).

Reply:
If the max value can't be used in a sum, then it's fine as is, but if it might be used, having this alternative value (I would use a named constant) that produces the correct behaviour would be safer.

Reply:
They (this + below) won't - both of these are only used in ComputeProbability and this will send any very large numbers caused by this into an else block.